test/module.test.js

```diff
@@ -1407,6 +1407,61 @@ describe('with router base', () => {
     expect(response.headers.location).toBe('/app/fr')
   })
 
+  test('detectBrowserLanguage redirects on non-base path', async () => {
+    const requestOptions = {
+      headers: {
+        'Accept-Language': 'fr'
+      }
+    }
+    const html = await get('/app/simple', requestOptions)
+    const dom = getDom(html)
+    expect(dom.querySelector('#container')?.textContent).toBe('Homepage')
+  })
+})
+
+describe('with router base + redirectOn is all', () => {
+  /** @type {Nuxt} */
+  let nuxt
+
+  beforeAll(async () => {
+    const override = {
+      router: {
+        base: '/app/'
+      },
+      i18n: {
+        detectBrowserLanguage: {
+          redirectOn: 'all'
+        }
+      }
+    }
+
+    nuxt = (await setup(loadConfig(__dirname, 'basic', override))).nuxt
```

Comment:
```suggestion
    nuxt = (await setup(loadConfig(__dirname, 'basic', override, { merge: true }))).nuxt
```

Reply:
Thank you! Looks like I was to hungry :)